src/exporter.py

```diff
@@ -210,7 +210,8 @@ def apply_config(self, exporter_config: Dict[str, Any]) -> None:
         logger.info("Updating exporter service configuration.")
         self.validate_config(exporter_config)
 
-        with open(self.SNAP_CONFIG_PATH, "w", encoding="utf-8") as config_file:
+        file_descriptor = os.open(self.SNAP_CONFIG_PATH, os.O_CREAT | os.O_WRONLY, 0o600)
```

Comment:
Will this change the permissions if the file already exists?  The python docs for os.open isn't very confident:

> Open the file path and set various flags according to flags and **possibly its mode according to mode.**

https://docs.python.org/3/library/os.html#os.open

Reply:
No, it won't. I've tested that before

Reply:
alternatively, we can use `os.chmod`

Reply:
+1 to os.chmod in that case - if we're changing the file permissions, it would be good for it to update the permissions on charm upgrade (when the file already exists).

Reply:
update :)